fluent-bundle/src/resource.rs

```diff
@@ -0,0 +1,40 @@
+use fluent_syntax::ast;
+use fluent_syntax::parser::parse;
+use fluent_syntax::parser::ParserError;
+
+rental! {
+    mod rentals {
+        use super::*;
+        #[rental(covariant, debug)]
+        pub struct FluentResource {
+            string: String,
+            ast: ast::Resource<'string>,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct FluentResource(rentals::FluentResource);
+
+impl FluentResource {
+    pub fn try_new(source: String) -> Result<Self, (Self, Vec<ParserError>)> {
```

Comment:
Can/should this take Into<String> for better caller ergonomics?

Reply:
We considered that with @manishearth and he recommended not doing that in order to avoid hiding the allocation.
While ergonomically that would be nice, it would also make it harder for the reader to understand that we will allocate the String and store it inside of the FluentResource.

In the future, when static `&str` will become implicitly convertible to String [0] it should be easier.

[0] https://github.com/rust-lang/rust/pull/46993

Reply:
Note: a thing that could be done here is to make the return value tuple into a proper public struct so it's easier to document